sklearn/gaussian_process/gpr.py

```diff
@@ -209,12 +219,22 @@ def obj_func(theta, eval_gradient=True):
                         "Multiple optimizer restarts (n_restarts_optimizer>0) "
                         "requires that all bounds are finite.")
                 bounds = self.kernel_.bounds
-                for iteration in range(self.n_restarts_optimizer):
-                    theta_initial = \
-                        self.rng.uniform(bounds[:, 0], bounds[:, 1])
-                    optima.append(
-                        self._constrained_optimization(obj_func, theta_initial,
-                                                       bounds))
+                # Initial theta values for 'n_restarts_optimizer' iterations
+                # for better optima
+                theta_initials = \
+                    self.rng.uniform(bounds[:, 0], bounds[:, 1],
+                                     size=(self.n_restarts_optimizer,
+                                     len(self.kernel_.theta)))
+                # 'n_restarts_optimizer' iterations using Embarrassingly
+                # Parallel helper
+                delayed_optima = delayed(self._optima_iterations,
+                                         check_pickle=False)
```

Comment:
Adding inline comments for non obvious changes like this can ease the review process. Of course it's better if you have a more detailed explanation rather than something like "it did not work for some reason, so I try something else and now it works, oh well"